Basic auth support for ES endpoints

preflightsiren opened the pull request:
simple addition to support basic authentication headers in the requests to elastic search.

resolves #10

Commit:
Adding basic auth username + password

Commit:
adding commons-codec to distribution

s12v commented on build.xml:
```diff
@@ -113,7 +117,8 @@
             <fileset dir="${main.build.classes.dir}"/>
             <zipfileset src="${sdk.jar}" />
-            <zipfileset src="${gson.jar}" />
+                <zipfileset src="${gson.jar}" />
```
extra indent

Commit:
Removing extra indenting

s12v commented:
@preflightsiren, thanks for the PR! Would it make sense to put empty username/password to the configuration template?

Commit:
Adding username/pass to config template

preflightsiren commented:
thanks for the quick review @s12v ! I've incorporated your suggestions (fixed some whitespace, added examples to the config template).

One improvement that could be made is to add to the JUnit tests to ensure that the authorization headers are only added when username and password is defined. I'm not super confident with JUnit so any help would be appreciated :D

preflightsiren commented:
@s12v was there anything else you wanted to see for this PR?

s12v commented:
@preflightsiren, no, thank you. I will test/merge it next week, I had very limited internet coverage this week :)

s12v commented:
@preflightsiren, thank you

jdavisclark commented:
@s12v: when will this be published to Plugin Central?
